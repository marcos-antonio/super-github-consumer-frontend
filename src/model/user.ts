export class User {
  constructor(user: Partial<User>) {
    this.id = user.id;
    this.login = user.login;
    this.htmlUrl = user.htmlUrl;
    this.avatarUrl = user.avatarUrl;
    this.createdAt = user.createdAt;
  }

  id: number;
  login: string;
  htmlUrl: string;
  avatarUrl: string;
  createdAt: string;
}
