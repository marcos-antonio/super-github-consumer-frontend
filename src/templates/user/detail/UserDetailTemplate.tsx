import React from 'react';

import * as M from './mock';
import * as S from './styled';

export const UserDetailTemplate = () => {
  const { login, id, url, created_at } = M.user;

  return (
    <S.Container>
      <S.FieldVisualizer>
        <S.FieldLabel>Login:</S.FieldLabel>
        <span>{login}</span>
      </S.FieldVisualizer>
      <S.FieldVisualizer>
        <S.FieldLabel>Id:</S.FieldLabel>
        <span>{id}</span>
      </S.FieldVisualizer>
      <S.FieldVisualizer>
        <S.FieldLabel>URL:</S.FieldLabel>
        <span>{url}</span>
      </S.FieldVisualizer>
      <S.FieldVisualizer>
        <S.FieldLabel>created_at:</S.FieldLabel>
        <span>{created_at}</span>
      </S.FieldVisualizer>
    </S.Container>
  );
};
