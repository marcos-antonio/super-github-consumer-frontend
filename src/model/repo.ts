import { User } from './user';

export class Repo {
  constructor(repo: Partial<Repo>) {
    this.id = repo.id;
    this.name = repo.name;
    this.htmlUrl = repo.htmlUrl;
    this.owner = repo.owner;
  }

  id: number;
  name: string;
  htmlUrl: string;

  owner: User;
}
