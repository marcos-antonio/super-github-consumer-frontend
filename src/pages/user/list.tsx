import React from 'react'

import { UserListTemplate } from 'templates/user/list'

const Index = () => <UserListTemplate />

export default Index
