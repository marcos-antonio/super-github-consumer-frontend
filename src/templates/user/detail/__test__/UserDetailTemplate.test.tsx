import React from 'react';
import { render, RenderResult } from '@testing-library/react';

import { ThemeProvider } from 'styled-components';

import theme from 'components/Theme';
import { User } from 'model/user';
import { Repo } from 'model/repo';

import { UserDetailTemplate } from '../UserDetailTemplate';

describe('Component: UserDetailTemplate', () => {
  let wrapper: RenderResult;

  const user: User = {
    id: 583231,
    login: 'octocat',
    htmlUrl: 'https://github.com/octocaat',
    avatarUrl: 'https://avatars3.githubusercontent.com/u/583231?v=4',
    createdAt: '2011-01-25T18:44:36Z',
  };
  const usersRepos: Repo[] = [
    new Repo({
      id: 132935648,
      name: 'boysenberry-repo-1',
      htmlUrl: 'https://github.com/octocat/boysenberry-repo-1',
      owner: new User({ id: 583231, login: 'octocat' }),
    }),
    new Repo({
      id: 18221276,
      name: 'git-consortium',
      htmlUrl: 'https://github.com/octocat/git-consortium',
      owner: new User({ id: 583231, login: 'octocat' }),
    }),
    new Repo({
      id: 20978623,
      name: 'hello-worId',
      htmlUrl: 'https://github.com/octocat/hello-worId',
      owner: new User({ id: 583231, login: 'octocat' }),
    }),
  ];

  beforeEach(() => {
    wrapper = render(
      <ThemeProvider theme={theme}>
        <UserDetailTemplate user={user} userRepos={usersRepos} />
      </ThemeProvider>
    );
  });

  it('render the component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
