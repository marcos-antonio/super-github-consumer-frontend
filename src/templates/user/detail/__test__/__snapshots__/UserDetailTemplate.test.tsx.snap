// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: UserDetailTemplate render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="styled__Container-sc-1cpfoq6-0 jtSYYf"
      >
        <div
          class="styled__OuterContainer-sc-1n0101z-0 esPeft styled__PaddedCard-sc-1cpfoq6-1 iflVqZ"
        >
          <div
            class="styled__AvatarContainer-sc-1cpfoq6-2 gtcBXx"
          >
            <img
              class="styled__StyledAvatar-sc-1cpfoq6-3 dcoiU"
              src="https://avatars3.githubusercontent.com/u/583231?v=4"
            />
          </div>
          <div
            class="styled__FieldVisualizerContainer-sc-1cpfoq6-4 dzCPpZ"
          >
            <div
              class="styled__FieldVisualizer-sc-1cpfoq6-5 brYMUh"
            >
              <span
                class="styled__FieldLabel-sc-1cpfoq6-6 dBnWVU"
              >
                Login
              </span>
              octocat
            </div>
          </div>
          <div
            class="styled__FieldVisualizerContainer-sc-1cpfoq6-4 fPafDS"
          >
            <div
              class="styled__FieldVisualizer-sc-1cpfoq6-5 brYMUh"
            >
              <span
                class="styled__FieldLabel-sc-1cpfoq6-6 dBnWVU"
              >
                User ID
              </span>
              583231
            </div>
          </div>
          <div
            class="styled__FieldVisualizerContainer-sc-1cpfoq6-4 dzCPpZ"
          >
            <div
              class="styled__FieldVisualizer-sc-1cpfoq6-5 brYMUh"
            >
              <span
                class="styled__FieldLabel-sc-1cpfoq6-6 dBnWVU"
              >
                Profile URL
              </span>
              <a
                href="https://github.com/octocaat"
                rel="noreferrer noopener"
                target="_blank"
              >
                https://github.com/octocaat
              </a>
            </div>
          </div>
          <div
            class="styled__FieldVisualizerContainer-sc-1cpfoq6-4 fPafDS"
          >
            <div
              class="styled__FieldVisualizer-sc-1cpfoq6-5 brYMUh"
            >
              <span
                class="styled__FieldLabel-sc-1cpfoq6-6 dBnWVU"
              >
                Created
              </span>
              25/1/2011
            </div>
          </div>
          <div
            class="styled__ReposTableContainer-sc-1cpfoq6-7 kryfrd"
          >
            <div
              class="styled__Container-q0w278-0 iFBbUj"
            >
              <div
                class="styled__Table-q0w278-1 eJXAqz"
              >
                <div
                  class="styled__HeaderContainer-q0w278-2 fUNRQo"
                >
                  <div
                    class="styled__Header-q0w278-3 klUxVQ"
                  >
                    <div
                      class="styled__HeaderCell-q0w278-4 cXSibf"
                    >
                      ID
                    </div>
                    <div
                      class="styled__HeaderCell-q0w278-4 cXSibf"
                    >
                      Name
                    </div>
                    <div
                      class="styled__HeaderCell-q0w278-4 cXSibf"
                    >
                      URL
                    </div>
                  </div>
                </div>
                <div
                  class="styled__Body-q0w278-5 gHKXfy"
                >
                  <div
                    class="styled__RowContainer-q0w278-6 bOruCu"
                  >
                    <div
                      class="styled__Row-q0w278-7 htLHt"
                    >
                      <div
                        class="styled__RowCell-q0w278-8 fTlsqw"
                      >
                        132935648
                      </div>
                      <div
                        class="styled__RowCell-q0w278-8 fTlsqw"
                      >
                        boysenberry-repo-1
                      </div>
                      <div
                        class="styled__RowCell-q0w278-8 fTlsqw"
                      >
                        https://github.com/octocat/boysenberry-repo-1
                      </div>
                    </div>
                  </div>
                  <div
                    class="styled__RowContainer-q0w278-6 bOruCu"
                  >
                    <div
                      class="styled__Row-q0w278-7 htLHt"
                    >
                      <div
                        class="styled__RowCell-q0w278-8 fTlsqw"
                      >
                        18221276
                      </div>
                      <div
                        class="styled__RowCell-q0w278-8 fTlsqw"
                      >
                        git-consortium
                      </div>
                      <div
                        class="styled__RowCell-q0w278-8 fTlsqw"
                      >
                        https://github.com/octocat/git-consortium
                      </div>
                    </div>
                  </div>
                  <div
                    class="styled__RowContainer-q0w278-6 bOruCu"
                  >
                    <div
                      class="styled__Row-q0w278-7 htLHt"
                    >
                      <div
                        class="styled__RowCell-q0w278-8 fTlsqw"
                      >
                        20978623
                      </div>
                      <div
                        class="styled__RowCell-q0w278-8 fTlsqw"
                      >
                        hello-worId
                      </div>
                      <div
                        class="styled__RowCell-q0w278-8 fTlsqw"
                      >
                        https://github.com/octocat/hello-worId
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="styled__PaginationContainer-q0w278-9 huNEmJ"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="styled__Container-sc-1cpfoq6-0 jtSYYf"
    >
      <div
        class="styled__OuterContainer-sc-1n0101z-0 esPeft styled__PaddedCard-sc-1cpfoq6-1 iflVqZ"
      >
        <div
          class="styled__AvatarContainer-sc-1cpfoq6-2 gtcBXx"
        >
          <img
            class="styled__StyledAvatar-sc-1cpfoq6-3 dcoiU"
            src="https://avatars3.githubusercontent.com/u/583231?v=4"
          />
        </div>
        <div
          class="styled__FieldVisualizerContainer-sc-1cpfoq6-4 dzCPpZ"
        >
          <div
            class="styled__FieldVisualizer-sc-1cpfoq6-5 brYMUh"
          >
            <span
              class="styled__FieldLabel-sc-1cpfoq6-6 dBnWVU"
            >
              Login
            </span>
            octocat
          </div>
        </div>
        <div
          class="styled__FieldVisualizerContainer-sc-1cpfoq6-4 fPafDS"
        >
          <div
            class="styled__FieldVisualizer-sc-1cpfoq6-5 brYMUh"
          >
            <span
              class="styled__FieldLabel-sc-1cpfoq6-6 dBnWVU"
            >
              User ID
            </span>
            583231
          </div>
        </div>
        <div
          class="styled__FieldVisualizerContainer-sc-1cpfoq6-4 dzCPpZ"
        >
          <div
            class="styled__FieldVisualizer-sc-1cpfoq6-5 brYMUh"
          >
            <span
              class="styled__FieldLabel-sc-1cpfoq6-6 dBnWVU"
            >
              Profile URL
            </span>
            <a
              href="https://github.com/octocaat"
              rel="noreferrer noopener"
              target="_blank"
            >
              https://github.com/octocaat
            </a>
          </div>
        </div>
        <div
          class="styled__FieldVisualizerContainer-sc-1cpfoq6-4 fPafDS"
        >
          <div
            class="styled__FieldVisualizer-sc-1cpfoq6-5 brYMUh"
          >
            <span
              class="styled__FieldLabel-sc-1cpfoq6-6 dBnWVU"
            >
              Created
            </span>
            25/1/2011
          </div>
        </div>
        <div
          class="styled__ReposTableContainer-sc-1cpfoq6-7 kryfrd"
        >
          <div
            class="styled__Container-q0w278-0 iFBbUj"
          >
            <div
              class="styled__Table-q0w278-1 eJXAqz"
            >
              <div
                class="styled__HeaderContainer-q0w278-2 fUNRQo"
              >
                <div
                  class="styled__Header-q0w278-3 klUxVQ"
                >
                  <div
                    class="styled__HeaderCell-q0w278-4 cXSibf"
                  >
                    ID
                  </div>
                  <div
                    class="styled__HeaderCell-q0w278-4 cXSibf"
                  >
                    Name
                  </div>
                  <div
                    class="styled__HeaderCell-q0w278-4 cXSibf"
                  >
                    URL
                  </div>
                </div>
              </div>
              <div
                class="styled__Body-q0w278-5 gHKXfy"
              >
                <div
                  class="styled__RowContainer-q0w278-6 bOruCu"
                >
                  <div
                    class="styled__Row-q0w278-7 htLHt"
                  >
                    <div
                      class="styled__RowCell-q0w278-8 fTlsqw"
                    >
                      132935648
                    </div>
                    <div
                      class="styled__RowCell-q0w278-8 fTlsqw"
                    >
                      boysenberry-repo-1
                    </div>
                    <div
                      class="styled__RowCell-q0w278-8 fTlsqw"
                    >
                      https://github.com/octocat/boysenberry-repo-1
                    </div>
                  </div>
                </div>
                <div
                  class="styled__RowContainer-q0w278-6 bOruCu"
                >
                  <div
                    class="styled__Row-q0w278-7 htLHt"
                  >
                    <div
                      class="styled__RowCell-q0w278-8 fTlsqw"
                    >
                      18221276
                    </div>
                    <div
                      class="styled__RowCell-q0w278-8 fTlsqw"
                    >
                      git-consortium
                    </div>
                    <div
                      class="styled__RowCell-q0w278-8 fTlsqw"
                    >
                      https://github.com/octocat/git-consortium
                    </div>
                  </div>
                </div>
                <div
                  class="styled__RowContainer-q0w278-6 bOruCu"
                >
                  <div
                    class="styled__Row-q0w278-7 htLHt"
                  >
                    <div
                      class="styled__RowCell-q0w278-8 fTlsqw"
                    >
                      20978623
                    </div>
                    <div
                      class="styled__RowCell-q0w278-8 fTlsqw"
                    >
                      hello-worId
                    </div>
                    <div
                      class="styled__RowCell-q0w278-8 fTlsqw"
                    >
                      https://github.com/octocat/hello-worId
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="styled__PaginationContainer-q0w278-9 huNEmJ"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
